libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1087,10 +1090,21 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 	{
 		// No initial value is only permitted for single variables with specified type.
 		if (_statement.declarations().size() != 1 || !_statement.declarations().front())
-			m_errorReporter.fatalTypeError(_statement.location(), "Assignment necessary for type detection.");
+		{
+			m_errorReporter.syntaxError(_statement.location(), "Use of the \"var\" keyword is disallowed.");
```

Comment:
Is this only the case for ``var``? If there is no test for it already, you should add one for:
```
(uint a, uint b, bytes c);
```
Typed multi-variable declarations are only allowed with assignment as well and I could imagine that they also end up being caught here, but I may be wrong. In any case the error message indeed has to be improved.

Reply:
@ekpyron This case is already caught by the parser (i.e. it generate a Parser error receiving a `;` but expecting an `=`). I however didn't find a test for this, so I added one in `/test/libsolidity/syntaxTests/parsing/tuples_decl_without_rhs.sol`.